provider/blob/src/export/blob_exporter.rs

```diff
@@ -41,31 +50,64 @@ impl DataExporter for BlobExporter<'_> {
             output: AllocVec::new(),
         };
         payload.serialize(&mut serializer)?;
-
-        let output = serializer.output;
+        let output = serializer
+            .output
+            .finalize()
+            .expect("Failed to finalize serializer output");
+        let idx = {
+            let mut unique_resources = self.unique_resources.lock().expect("poison");
+            let len = unique_resources.len();
+            *unique_resources.entry(output).or_insert(len)
+        };
         #[allow(clippy::expect_used)]
         self.resources.lock().expect("poison").push((
             key.get_hash(),
             options.write_to_string().into_owned().into_bytes(),
-            output
-                .finalize()
-                .expect("Failed to finalize serializer output"),
+            idx,
         ));
         Ok(())
     }
 
     fn close(&mut self) -> Result<(), DataError> {
-        #[allow(clippy::expect_used)]
+        // The blob IDs are unstable due to the parallel nature of datagen.
```

Comment:
thought: hmmm, i bet this means that the binary diff will be large on each datagen, which could potentially be a problem for git repo sizes (but it's quite likely these files are already not using packfiles)

Reply:
This means that new data blobs will be inserted into a predictable place, and the ordering is stable